src/actions/player/index.js

```diff
@@ -90,12 +97,31 @@ function getIteratedTrack(playlist, currentActiveTrackId, iterate) {
   return playlist[index + iterate];
 }
 
+function getRandomTrack(playlist, currentActiveTrackId) {
+  const index = findIndex(isSameTrack(currentActiveTrackId), playlist);
+  function getRandomIndex() {
+    const randNum = Math.floor(Math.random() * playlist.length);
+    if (randNum === index) {
+      return getRandomIndex();
+    } else {
+      return randNum;
+    }
+  }
+  return playlist[getRandomIndex()];
+}
+
 export const activateIteratedTrack = (currentActiveTrackId, iterate) => (dispatch, getState) => {
   const playlist = getState().player.playlist;
   const nextActiveTrackId = getIteratedTrack(playlist, currentActiveTrackId, iterate);
+  const randomActiveTrackId = getRandomTrack(playlist, currentActiveTrackId);
+  const shuffleMode = getState().player.shuffleMode;
+  // const previousActiveTrackId = getState().player.activeTrackId;
+
 
-  if (nextActiveTrackId) {
+  if (nextActiveTrackId && shuffleMode === false) {
     dispatch(activateTrack(nextActiveTrackId));
+  } else if (shuffleMode === true) {
+    dispatch(activateTrack(randomActiveTrackId));
```

Comment:
It would be sufficient to calculate the `randomActiveTrackId ` only when you get in that if block.